enginetest/queries/script_queries.go

```diff
@@ -138,6 +138,125 @@ var ScriptTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "intersection tests",
```

Comment:
i noted below that RECURSIVE CTE should be its own testing block, because of the way we are overloading the nodes, and I don't know whether MySQL supports including INTERSECT/EXCEPT in their special union handling.

Reply:
need join testing. joins with INTERSECT/EXCEPT as both LHS and RHS. Joins with subqueries that have INTERSECT/EXCEPT. ideally in join_op_tests, so that we run them with all of the execution operators

Reply:
nested EXCEPT/INTERCEPT seems like an initial layer of testing we have to have also. plus set arith in subqueries and subquery expressions to make sure we index them correctly

Reply:
not seeing LIMIT, GROUP BY, HAVING tests

Reply:
maybe some lateral join tests